### PR TITLE
Prettier config and workflow setup

### PR DESCRIPTION
**Pull request by to-sta**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

This PR is related to the recently merged PR #324. It builds on the GitHub workflow but instead of using a predefined Github Action `creyD/prettier_action@v4.3` we manually install prettier an run it. 

Prettier was installed via npm and configuration files were added. 

I ran a test of prettier and the GitHub workflow with act. Both worked and only [warn] showed up, where most of changes are the end of line sequence from `crlf` to `lf`. Prettier uses `lf` by default, we could overwrite the default in the `.prettierrc`.

```json
"endOfLine": "crlf"
```
I did not run `npx prettier . --write` to make the changes yet.

### Related issue

- #323

### CODE REVIEW

File: frontend/.prettierrc
Comment (andrewtavis) [inline frontend/.prettierrc:2]: What are we doing for trailing commas with this? In looking at it a little bit more, I think that I might have been convinced that my hereto dislike of trailing commas at the end of lists and arrays might be unwarranted. Might be preferable even as it'd give us nicer git diffs and also would making extending lists easier (I've not enjoyed all the JSON work with the L10N files where I need to add the comma myself all the time). How do you feel about enforcing trailing comma addition?

File: frontend/.prettierrc
Comment (andrewtavis) [inline frontend/.prettierrc:5]: Yessss to all of the above :) :)

**Comment by andrewtavis** [comment]
And doing a bit of research, `crlf` is two bytes instead of one, so I'm ok with keeping the prettier default. It would just be a matter of those of us on Mac and Linux going through and removing/adding in the end line again in some cases to make sure that we get the same type of break?

File: frontend/.prettierrc
Comment (to-sta) [reply frontend/.prettierrc:2]: I don't have a strong opinion on this. Personally i do not write trailing commas and let the formatter take care of it. My default setting in VScode was also "es5".

Just checked the docs for the default and had changed:

> Default value changed from es5 to all in v3.0.0

I am happy with any of the three options.

Comment (andrewtavis) [reply frontend/.prettierrc:2]: So as of now is it setup to add the last comma to it, @to-sta? If not, can that be swapped in a commit? Happy to merge once that's done :) :)

Comment (to-sta) [reply frontend/.prettierrc:2]: Yeah, it will add a trailing comma.

Comment (andrewtavis) [reply frontend/.prettierrc:2]: Thanks, @to-sta :) Has been a long day of traveling and just wanted to check :)

**Review by andrewtavis — APPROVED** [review]
Thanks for all the work here, @to-sta! Merging this in, and let's see how the next PRs interact with it 😊 Am realizing as well that there's now a `package-lock.json` file here, which I'll remove quickly after merging :)

File: .vscode/settings.json
Comment (wkyoshida) [inline .vscode/settings.json:4]: Sorry, real quick - was looking at some of the `prettier` additions.

Could having some of these configs in the `.vscode` configs cause some issues? Some individuals may not be using VSCode. Or if they are, could this conflict with any `settings.json` files that folks may already have? Just wondering

Comment (to-sta) [reply .vscode/settings.json:4]: Added an issue for it #363 and will work on this in the next days😃

**Pull request merged**
